### PR TITLE
[Move] Implement Safeguard

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
Implements the move Safeguard.

## Why am I doing these changes?
Implement more moves.

## What did change?
Creates an `ArenaTag` used to implement the effects of Safeguard. Checks for the `ArenaTag` are implemented in `ConfuseAttr` and `canSetStatus()`.

## How to test the changes?
`npm run test safeguard`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [ ] There is no overlap with another PR?
  - Updated version of #3108 
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/locales/fr/move-trigger.ts [hunk 1]
```diff
@@ -61,4 +61,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "suppressAbilities": "Le talent de {{pokemonName}}\na été rendu inactif !",
   "swapArenaTags": "Les effets affectant chaque côté du terrain\nont été échangés par {{pokemonName}} !",
   "exposedMove": "{{targetPokemonName}} est identifié\npar {{pokemonName}} !",
+  "safeguard": "{{targetName}} is protected by Safeguard!",
```
Comment (Adri1) [inline src/locales/fr/move-trigger.ts:64]:
```suggestion
  "safeguard": "{{targetName}} est protégé\npar la capacité Rune Protect !",
```

File: src/locales/fr/arena-tag.ts [hunk 2]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "Le vent arrière soufflant\nsur l’équipe ennemie s’arrête !",
   "happyHourOnAdd": "L’ambiance est euphorique !",
   "happyHourOnRemove": "L’ambiance se calme !",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (Adri1) [inline src/locales/fr/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "Un voile mystérieux recouvre\ntout le terrain !",
  "safeguardOnAddPlayer": "Un voile mystérieux recouvre\nvotre équipe !",
  "safeguardOnAddEnemy": "Un voile mystérieux recouvre\nl’équipe ennemie !",
  "safeguardOnRemove": "Le terrain n’est plus protégé\npar le voile mystérieux !",
  "safeguardOnRemovePlayer": "Votre équipe n’est plus protégée\npar le voile mystérieux !",
  "safeguardOnRemoveEnemy": "L’équipe ennemie n’est plus protégée\npar le voile mystérieux !",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/pt_BR/move-trigger.ts [hunk 3]
```diff
@@ -61,4 +61,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "suppressAbilities": "A habilidade de {{pokemonName}}\nfoi suprimida!",
   "swapArenaTags": "{{pokemonName}} trocou os efeitos de batalha que afetam cada lado do campo!",
   "exposedMove": "{{pokemonName}} identificou\n{{targetPokemonName}}!",
+  "safeguard": "{{targetName}} is protected by Safeguard!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/move-trigger.ts:64]:
```suggestion
  "safeguard": "{{targetName}} está protegido por Safeguard!",
```

File: src/locales/pt_BR/arena-tag.ts [hunk 4]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "O Tailwind da equipe adversária acabou!",
   "happyHourOnAdd": "Todos foram envolvidos por uma atmosfera alegre!",
   "happyHourOnRemove": "A atmosfera retornou ao normal.",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "O campo de batalha está envolto num véu místico!",
  "safeguardOnAddPlayer": "Sua equipe se envolveu num véu místico!",
  "safeguardOnAddEnemy": "A equipe adversária se envolveu num véu místico!",
  "safeguardOnRemove": "O campo não está mais protegido por Safeguard!",
  "safeguardOnRemovePlayer": "Sua equipe não está mais protegido por Safeguard!",
  "safeguardOnRemoveEnemy": "A equipe adversária não está mais protegido por Safeguard!",
```

File: src/locales/zh_TW/arena-tag.ts [hunk 5]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "The opposing team's Tailwind petered out!",
   "happyHourOnAdd": "Everyone is caught up in the happy atmosphere!",
   "happyHourOnRemove": "The atmosphere returned to normal.",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "整個場地被\n神秘之幕包圍了！",
  "safeguardOnAddPlayer": "我方被\n神秘之幕包圍了！",
  "safeguardOnAddEnemy": "對手被\n神秘之幕包圍了！",
  "safeguardOnRemove": "包圍整個場地的\n神秘之幕消失了！",
  "safeguardOnRemovePlayer": "包圍我方的\n神秘之幕消失了！",
  "safeguardOnRemoveEnemy": "包圍對手的\n神秘之幕消失了！",
```

File: src/locales/zh_CN/arena-tag.ts [hunk 6]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "敌方的顺风停止了！",
   "happyHourOnAdd": "大家被欢乐的\n气氛包围了！",
   "happyHourOnRemove": "气氛回复到平常了。",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (mercurius-00) [inline src/locales/zh_CN/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "整个场地被\n神秘之幕包围了！",
  "safeguardOnAddPlayer": "我方被\n神秘之幕包围了！",
  "safeguardOnAddEnemy": "对手被\n神秘之幕包围了！",
  "safeguardOnRemove": "包围整个场地的\n神秘之幕消失了！",
  "safeguardOnRemovePlayer": "包围我方的\n神秘之幕消失了！",
  "safeguardOnRemoveEnemy": "包围对手的\n神秘之幕消失了！",
```

File: src/locales/de/arena-tag.ts [hunk 7]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "Der Rückenwind auf gegnerischer Seite hat sich gelegt!",
   "happyHourOnAdd": "Goldene Zeiten sind angebrochen!",
   "happyHourOnRemove": "Die goldenen Zeiten sind vorbei!",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (CodeTappert) [inline src/locales/de/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "Das ganze Feld wird von einem Schleier umhüllt!",
  "safeguardOnAddPlayer": "Das Team des Anwenders wird von einem Schleier umhüllt!",
  "safeguardOnAddEnemy": "Das gegnerische Team wird von einem Schleier umhüllt!",
  "safeguardOnRemove": "Der mystische Schleier, der das ganze Feld umgab, hat sich gelüftet!",
  "safeguardOnRemovePlayer": "Der mystische Schleier, der dein Team umgab, hat sich gelüftet!",
  "safeguardOnRemoveEnemy": "Der mystische Schleier, der das gegnerische Team umgab, hat sich gelüftet!",
```

File: src/locales/de/move-trigger.ts [hunk 8]
```diff
@@ -61,4 +61,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "suppressAbilities": "Die Fähigkeit von {{pokemonName}} wirkt nicht mehr!",
   "swapArenaTags": "{{pokemonName}} hat die Effekte, die auf den beiden Seiten des Kampffeldes wirken, miteinander getauscht!",
   "exposedMove": "{{pokemonName}} erkennt {{targetPokemonName}}!",
+  "safeguard": "{{targetName}} is protected by Safeguard!",
```
Comment (CodeTappert) [inline src/locales/de/move-trigger.ts:64]:
```suggestion
  "safeguard": "{{targetName}} wird durch Bodyguard geschützt!",
```

File: src/locales/es/arena-tag.ts [hunk 9]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "The opposing team's Tailwind petered out!",
   "happyHourOnAdd": "Everyone is caught up in the happy atmosphere!",
   "happyHourOnRemove": "The atmosphere returned to normal.",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (AsdarDevelops) [inline src/locales/es/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "¡Todos los Pokémon están protegidos por Velo Sagrado!",
  "safeguardOnAddPlayer": "¡Tu equipo se ha protegido con Velo Sagrado!",
  "safeguardOnAddEnemy": "¡El equipo enemigo se ha protegido con Velo Sagrado!",
  "safeguardOnRemove": "¡Velo Sagrado dejó de hacer efecto!",
  "safeguardOnRemovePlayer": "El efecto de Velo Sagrado en tu equipo se ha disipado.",
  "safeguardOnRemoveEnemy": "El efecto de Velo Sagrado en el equipo enemigo
se ha disipado.",
```

File: src/locales/es/move-trigger.ts [hunk 10]
```diff
@@ -61,4 +61,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
   "swapArenaTags": "{{pokemonName}} swapped the battle effects affecting each side of the field!",
   "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
+  "safeguard": "{{targetName}} is protected by Safeguard!",
```
Comment (AsdarDevelops) [inline src/locales/es/move-trigger.ts:64]:
```suggestion
  "safeguard": "¡{{targetName}} está protegido por Velo Sagrado!",
```

**Review by AsdarDevelops — COMMENTED** [review]
Added Spanish

File: src/locales/ko/move-trigger.ts [hunk 11]
```diff
@@ -61,4 +61,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "suppressAbilities": "{{pokemonName}}의\n특성이 효과를 발휘하지 못하게 되었다!",
   "swapArenaTags": "{{pokemonName}}[[는]]\n서로의 필드 효과를 교체했다!",
   "exposedMove": "{{pokemonName}}[[는]]\n{{targetPokemonName}}의 정체를 꿰뚫어 보았다!",
+  "safeguard": "{{targetName}} is protected by Safeguard!",
```
Comment (sodaMelon) [inline src/locales/ko/move-trigger.ts:64]:
```suggestion
  "safeguard": "{{targetName}}[[는]] 신비의 베일이 지켜 주고 있다!",
```

File: src/locales/ko/arena-tag.ts [hunk 12]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "상대의\n순풍이 멈췄다!",
   "happyHourOnAdd": "모두 행복한 기분에\n휩싸였다!",
   "happyHourOnRemove": "기분이 원래대로 돌아왔다.",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (sodaMelon) [inline src/locales/ko/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "필드 전체가 신비의 베일에 둘러싸였다!",
  "safeguardOnAddPlayer": "우리 편은 신비의 베일에 둘러싸였다!",
  "safeguardOnAddEnemy": "상대 편은 신비의 베일에 둘러싸였다!",
  "safeguardOnRemove": "필드를 감쌌던 신비의 베일이 없어졌다!",
  "safeguardOnRemovePlayer": "우리 편을 감싸던 신비의 베일이 없어졌다!",
  "safeguardOnRemoveEnemy": "상대를 감싸던 신비의 베일이 없어졌다!",
```

**Review by sodaMelon — COMMENTED** [review]
added korean localization

File: src/locales/it/move-trigger.ts [hunk 13]
```diff
@@ -62,4 +62,5 @@ export const moveTriggers: SimpleTranslationEntries = {
   "suppressAbilities": "L’abilità di {{pokemonName}}\nperde ogni efficacia!",
   "swapArenaTags": "{{pokemonName}} ha invertito gli effetti attivi\nnelle due metà del campo!",
   "exposedMove": "{{pokemonName}} identified\n{{targetPokemonName}}!",
+  "safeguard": "{{targetName}} is protected by Safeguard!",
```
Comment (NicusPulcis) [inline src/locales/it/move-trigger.ts:65]:
```suggestion
  "exposedMove": "{{targetPokemonName}} è stato identificato\n da {{pokemonName}}!",
  "safeguard": "Salvaguardia protegge {{targetName}}!",
```

File: src/locales/it/arena-tag.ts [hunk 14]
```diff
@@ -50,4 +50,10 @@ export const arenaTag: SimpleTranslationEntries = {
   "tailwindOnRemoveEnemy": "The opposing team's Tailwind petered out!",
   "happyHourOnAdd": "Everyone is caught up in the happy atmosphere!",
   "happyHourOnRemove": "The atmosphere returned to normal.",
+  "safeguardOnAdd": "The whole field is cloaked in a mystical veil!",
+  "safeguardOnAddPlayer": "Your team cloaked itself in a mystical veil!",
+  "safeguardOnAddEnemy": "The opposing team cloaked itself in a mystical veil!",
+  "safeguardOnRemove": "The field is no longer protected by Safeguard!",
+  "safeguardOnRemovePlayer": "Your team is no longer protected by Safeguard!",
+  "safeguardOnRemoveEnemy": "The opposing team is no longer protected by Safeguard!",
```
Comment (NicusPulcis) [inline src/locales/it/arena-tag.ts:58]:
```suggestion
  "safeguardOnAdd": "Un velo mistico ricopre il campo!",
  "safeguardOnAddPlayer": "Un velo mistico ricopre la tua squadra!",
  "safeguardOnAddEnemy": "Un velo mistico ricopre la squadra avversaria!",
  "safeguardOnRemove": "Il campo non è più protetto da Salvaguardia!",
  "safeguardOnRemovePlayer": "La tua squadra non è più protetta da Salvaguardia!",
  "safeguardOnRemoveEnemy": "La squadra avversaria non è più protetta da Salvaguardia!",
```

**Review by NicusPulcis — DISMISSED** [review]
Italian localization is done

File: src/data/move.ts [hunk 15]
```diff
@@ -4484,6 +4491,15 @@ export class ConfuseAttr extends AddBattlerTagAttr {
   constructor(selfTarget?: boolean) {
     super(BattlerTagType.CONFUSED, selfTarget, false, 2, 5);
   }
+
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
+    if (!this.selfTarget && target.scene.arena.getTagOnSide(ArenaTagType.SAFEGUARD, target.isPlayer() ? ArenaTagSide.PLAYER : ArenaTagSide.ENEMY)) {
+      user.scene.queueMessage(i18next.t("moveTriggers:safeguard", { targetName: getPokemonNameWithAffix(target)}));
```
Comment (innerthunder) [inline src/data/move.ts:4497]:
```suggestion
      if (move.category === MoveCategory.STATUS) {
        user.scene.queueMessage(i18next.t("moveTriggers:safeguard", { targetName: getPokemonNameWithAffix(target)}));
      }
```
So attacks like Hurricane can't trigger the Safeguard message

**Review by innerthunder — DISMISSED** [review]
lgtm 🚀

**Comment by frutescens** [comment]
Could you resolve conflicts?

**Pull request merged**